### PR TITLE
Add warning about regex experession in ML

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
### Description
_Describe what this change achieves._

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _ml-commons-plugin/cluster-settings.md [hunk 1]
```diff
@@ -143,8 +143,12 @@ The default value allows you to upload a model file from any http/https/ftp/loca
 
 ### Setting
 
+The default URL value for this trusted URL setting is not secure. To ensure the security, please use you own regex string to the repostory that contains your models, for example `https://github.com/opensearch-project/ml-commons/blob/2.x/ml-algorithms/src/test/resources/org/opensearch/ml/engine/algorithms/text_embedding/*`.
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/cluster-settings.md:146]:
```suggestion
The default URL value for this trusted URL setting is not secure. To ensure the security, please use you own regex string to the trusted repository that contains your models, for example `https://github.com/opensearch-project/ml-commons/blob/2.x/ml-algorithms/src/test/resources/org/opensearch/ml/engine/algorithms/text_embedding/*`.
```

File: _ml-commons-plugin/cluster-settings.md [hunk 2]
```diff
@@ -143,8 +143,12 @@ The default value allows you to upload a model file from any http/https/ftp/loca
...
+{: .warning }
+
+
 ```
-plugins.ml_commons.trusted_url_regex: ^(https?\|ftp\|file)://[-a-zA-Z0-9+&@#/%?=~_\|!:,.;]*[-a-zA-Z0-9+&@#/%=~_\|]
+plugins.ml_commons.trusted_url_regex: <your-model-url>
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/cluster-settings.md]:
```suggestion
plugins.ml_commons.trusted_url_regex: "^(https?|ftp|file)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]"
```

File: _ml-commons-plugin/cluster-settings.md [hunk 3]
```diff
@@ -143,8 +143,12 @@ The default value allows you to upload a model file from any http/https/ftp/loca
 
 ### Setting
 
+The default URL value for this trusted URL setting is not secure. To ensure the security, please use you own regex string to the repostory that contains your models, for example `https://github.com/opensearch-project/ml-commons/blob/2.x/ml-algorithms/src/test/resources/org/opensearch/ml/engine/algorithms/text_embedding/*`.
+{: .warning }
+
+
 ```
-plugins.ml_commons.trusted_url_regex: ^(https?\|ftp\|file)://[-a-zA-Z0-9+&@#/%?=~_\|!:,.;]*[-a-zA-Z0-9+&@#/%=~_\|]
+plugins.ml_commons.trusted_url_regex: <model-repository-url>
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/cluster-settings.md:151]: Default value should be `"^(https?|ftp|file)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]"`

Comment (Naarcha-AWS) [reply _ml-commons-plugin/cluster-settings.md:151]: The default value is called out below. I'm ok with switching it back, but I figured we would try and de-emphasize the default value for this setting to encourage people to change it.

Comment (ylwu-amzn) [reply _ml-commons-plugin/cluster-settings.md:151]: Sure, I mean the default value of line 156 is not correct. I can't comment on line 156, so just add comment here.

**Review by ylwu-amzn — APPROVED** [review]

**Review by carolxob — APPROVED** [review]
LGTM.

**Review by hdhalter — APPROVED** [review]

**Pull request merged**